src/OpenTracing/ISpan.cs

```diff
@@ -1,164 +1,177 @@
-        /// <returns>The current <see cref="ISpan"/> instance for chaining.</returns>
+        /// <returns>This Span instance, for chaining</returns>
         ISpan SetBaggageItem(string key, string value);
 
         /// <summary>
-        /// Returns the value of the baggage item identified by the given <paramref name="key"/>,
-        /// or <c>null</c> if no such item could be found.
+        ///     The value of the baggage item identified by <paramref name="key" />, or null if no such item could be found.
         /// </summary>
-        /// <param name="key">The name of the key which was used to store the baggage item.</param>
-        string GetBaggageItem(string key);
+        string GetBaggaggeItem(string key);
+
+        /// <summary>
+        ///     Sets the string name for the logical operation this Span represents.
+        /// </summary>
+        /// <returns>This Span instance, for chaining</returns>
+        ISpan SetOperationName(string operationName);
 
         /// <summary>
-        /// <para>Sets the end timestamp to now and records the span.</para>
-        /// <para>
-        /// With the exception of calls to <see cref="Context"/> (which are always allowed),
-        /// this should be the last call made to the span instance, and to do otherwise
-        /// leads to undefined behavior.
-        /// </para>
+        ///     Sets the end timestamp to now and records the span.
+        ///     <para>
+        ///         With the exception of calls to <see cref="ISpan.Context" />, this should be the last call made to the
+        ///         span instance.
+        ///         Future calls to <see cref="Finish()" /> are defined as noops, and future calls to methods other than
+        ///         <see cref="ISpan.Context" />
+        ///         lead to undefined behavior (likely an exception).
+        ///     </para>
         /// </summary>
+        /// <seealso cref="ISpan.Context" />
         void Finish();
 
         /// <summary>
-        /// <para>Sets an explicit end timestamp and records the span.</para>
-        /// <para>
-        /// With the exception of calls to <see cref="Context"/> (which are always allowed),
-        /// this should be the last call made to the span instance, and to do otherwise
-        /// leads to undefined behavior.
-        /// </para>
-        /// <param name="finishTimestamp">An explicit finish timestamp.</param>
+        ///     Sets an explicit end timestamp and records the span.
+        ///     <para>
+        ///         With the exception of calls to Span.context(), this should be the last call made to the span instance, and to
+        ///         do otherwise leads to undefined behavior.
+        ///     </para>
         /// </summary>
-        void Finish(DateTimeOffset finishTimestamp);
+        /// <param name="finishTime">An explicit finish time</param>
+        /// <seealso cref="ISpan.Context" />
+        void Finish(DateTimeOffset finishTime);
```

Comment:
everything else uses "timestamp" so this should stay "finishTimestamp" IMO

Reply:
resolved